core/Network/TLS/Struct.hs

```diff
@@ -566,3 +586,18 @@ instance EnumSafe16 Group where
     toEnumSafe16 259 = Just FFDHE6144
     toEnumSafe16 260 = Just FFDHE8192
     toEnumSafe16 _   = Nothing
+
+
+-- Digital signature algorithm, in close relation to public/private key types
+-- and cipher key exchange.
+data DigitalSignatureAlg = RSA | DSS | ECDSA | Ed25519 | Ed448
+                           deriving (Show, Eq)
+
```

Comment:
Ah. In my old implementation, `signatureCompatible` is necessary for the `Credential` module which causes cyclic import. But this is not the case anymore. I will try.